engine/src/main/java/io/seldon/engine/service/InternalPredictionService.java

```diff
@@ -495,27 +496,7 @@ private SeldonMessage queryREST(
           logger.debug(response);
           SeldonMessage.Builder builder = SeldonMessage.newBuilder();
           JsonFormat.parser().ignoringUnknownFields().merge(response, builder);
-          SeldonMessage seldonMessage = builder.build();
-          if (httpResponse.getStatusCode().is2xxSuccessful()) {
-            return seldonMessage;
-          } else {
-            logger.error(
-                "Couldn't retrieve prediction from external prediction server -- bad http return code: "
-                    + httpResponse.getStatusCode());
-            Status seldonMessageStatus = seldonMessage.getStatus();
-            if (seldonMessageStatus == null) {
-                throw new APIException(
-                        APIException.ApiExceptionType.ENGINE_MICROSERVICE_ERROR,
-                        String.format("Bad return code %d", httpResponse.getStatusCodeValue()));
-            }
-            else
-            {
-                throw new APIException(seldonMessageStatus.getCode(),
-                                       seldonMessageStatus.getReason(),
-                                       200,
-                                       seldonMessageStatus.getInfo());
-            }
-          }
```

Comment:
Based on what was discussed in #1075, was it the case that this whole `else {...}` block was never reached?

If it was, we may be removing some functionality with this PR, since the new `handleHttpStatusCodeError(...)` method is only called when `restTemplate.postForEntity(...)` throws a `HttpStatusCodeException` exception.

Reply:
@adriangonz My local test shows any non-200 response makes the code path going to [this exception handling](https://github.com/SeldonIO/seldon-core/blob/master/engine/src/main/java/io/seldon/engine/service/InternalPredictionService.java#L534). I dig into the code, looks like `RestTemplate` class builder is using `ExtractingResponseErrorHandler` as its error handler, and it throws `RestClientException`with non-200 response. 

The inheriting relationship is `HttpServerErrorException` -> (meaning extends) `HttpStatusCodeException` -> `RestClientResponseException` -> `RestClientException`. So I extracted the `HttpStatusCodeException` here by moving the logic inside, while leaving all the other exceptions handled under `Exception` category.

Reply:
Got it! Thanks for the clarification!